File: w09/d03/jorge_colindres/mail_objects_app/app/assets/javascripts/mailbox.js
function Mailbox(){
  this.mailitems = [];
}

Mailbox.prototype.acceptMail = function(mail) {
  this.mailitems.push(mail);
};

Mailbox.prototype.openMail = function() {
  this.mailitems.shift();
};

Mailbox.prototype.toString = function() {
  var postcards = 0, packages = 0;
  for (var i = 0; i < this.mailitems.length; i++) {
    if (this.mailitems[i].contents) {
      packages++;
    } else if (this.mailitems[i].message){
      postcards++;
    }
  }
  return 'The mailbox has ' + postcards + ' postcards and ' + packages + ' packages.';
};

File: w09/d01/edward_shin/mail_it/mailbox.rb
require_relative 'postcard'
require_relative 'package'

class Mailbox

  def initialize
    @check_mail = []
  end

  def check_mail
    @check_mail
  end

  def accept_mail(mail)
    @check_mail << mail
  end
  def open_mail
    return @check_mail.shift
  end

  def to_s
    packages = 0
    postcards = 0
    @check_mail.each do |mail|
      if mail.class == Package
        packages += 1
      elsif mail.class == Postcard
        postcards += 1
      end
    end
    return "the mailbox has #{packages} packages and #{postcards} postcards"
  end

end